packages/xrpl/src/models/transactions/oracleSet.ts

```diff
@@ -146,7 +148,7 @@ export function validateOracleSet(tx: Record<string, unknown>): void {
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- we are validating the type
         'AssetPrice' in priceData.PriceData &&
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- we are validating the type
-        !isNumber(priceData.PriceData.AssetPrice)
+        !(isNumber(priceData.PriceData.AssetPrice) || (typeof priceData.PriceData.AssetPrice === 'string' && HEX_REGEX.test(priceData.PriceData.AssetPrice)))
```

Comment:
This regex test should potentially be included in `isNumber`

Reply:
There are many `unsigned int` values which aren't `std::uint64_t` amongst `sfABCD` values. Such values do not accept a `str` representation. There are 37 usages of this method in `packages/xrpl/`. Apart from a case-by-case decision, I don't know of a sweeping way to disambiguate this usage.

Reply:
Is that just true of the library or does rippled also not accept hex for smaller int values?

Reply:
Based on a perfunctory glance at the code base, rippled does not have any constructors which convert a hexa-decimal string into a `std::uint<>_t` type. All of them accept a `std::uint<>_t` type or a byte-string.

c++ code does not need a `hexadecimal string -> UINT` conversion because the `std::uint<>_t` types sufficiently represent the values. My understanding is that (non-c++) client libraries rely on strings to preserve the precision of values.

Source: https://github.com/XRPLF/rippled/blob/af018c7b0bd2d019a009e4405cfb7c3923c3a078/include/xrpl/protocol/STInteger.h#L38